Framework/Algorithms/test/FilterEventsTest.h

```diff
@@ -736,27 +732,31 @@ class FilterEventsTest : public CxxTest::TestSuite {
    * nanosecond
     * @param pulsedt : pulse length in int64_t format with unit nanosecond
     */
-  DataObjects::EventWorkspace_sptr
-  createEventWorkspaceInDirect(int64_t runstart_i64, int64_t pulsedt) {
+  EventWorkspace_sptr createEventWorkspaceInDirect(int64_t runstart_i64,
+                                                   int64_t pulsedt) {
     // Create an EventWorkspace with 10 banks with 1 detector each.  No events
     // is generated
-    DataObjects::EventWorkspace_sptr eventWS =
+    EventWorkspace_sptr eventWS =
         WorkspaceCreationHelper::createEventWorkspaceWithFullInstrument(10, 1,
                                                                         true);
-
     // Add EFixed to each detector
     const ParameterMap &pmap = eventWS->constInstrumentParameters();
 
     for (size_t i = 0; i < 10; ++i) {
-      Geometry::IDetector_const_sptr det = eventWS->getDetector(i);
-      Parameter_sptr par = pmap.getRecursive(det.get(), "Efixed");
+
+      const auto &spectrumInfo = eventWS->spectrumInfo();
```

Comment:
This was one of my doubts: I initially put `const auto &spectrumInfo = ...` outside the loop, but the test would crash. Is it also because `spectrumInfo` is being invalidated? Your comment below is related to this: I was trying to figure out what was happening (I'll remove it).